pkg/templates/pdb/template.go

```diff
@@ -0,0 +1,114 @@
+package pdb
+
+import (
+	"fmt"
+
+	"golang.stackrox.io/kube-linter/pkg/check"
+	"golang.stackrox.io/kube-linter/pkg/config"
+	"golang.stackrox.io/kube-linter/pkg/diagnostic"
+	"golang.stackrox.io/kube-linter/pkg/lintcontext"
+	"golang.stackrox.io/kube-linter/pkg/objectkinds"
+	"golang.stackrox.io/kube-linter/pkg/templates"
+	"golang.stackrox.io/kube-linter/pkg/templates/pdb/internal/params"
+	v1 "k8s.io/api/apps/v1"
+	pdbV1 "k8s.io/api/policy/v1"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+const (
+	templateKey           = "pod-disruption-budget"
+	maxUnavailableZeroMsg = "MaxUnavailable is set to 0"
+)
+
+func init() {
+	templates.Register(check.Template{
+		HumanName:   "Pod Disruption Budget Configuration Checks",
+		Key:         templateKey,
+		Description: "Flag non-optimal PDB configurations",
+		SupportedObjectKinds: config.ObjectKindsDesc{
+			ObjectKinds: []string{
+				objectkinds.PodDisruptionBudget},
+		},
+		Parameters:             params.ParamDescs,
+		ParseAndValidateParams: params.ParseAndValidate,
+		Instantiate: params.WrapInstantiateFunc(func(p params.Params) (check.Func, error) {
+			return func(lintCtx lintcontext.LintContext, object lintcontext.Object) []diagnostic.Diagnostic {
+
+				var results []diagnostic.Diagnostic
+
+				// Get the PDB provided
+				pdb, ok := object.K8sObject.(*pdbV1.PodDisruptionBudget)
+				if !ok {
+					return nil
+				}
+
+				// If MaxUnavailable is set to 0,
+				// return as no other checks will uniquely apply
+				if pdb.Spec.MaxUnavailable.IntVal == 0 {
+					return []diagnostic.Diagnostic{{
+						Message: maxUnavailableZeroMsg,
+					},
+					}
```

Comment:
nit in terms of formatting:
```suggestion
					return []diagnostic.Diagnostic{
					        {
						        Message: maxUnavailableZeroMsg,
					        },
					}
```